### ISSUE TITLE
1.6.9 regression: failed passive exchange declare causes InvalidConnectionState

### ISSUE THREAD

**Issue by tl-alex-butler** [CleverCloud/lapin#330]
I recently encountered the titular regression in my test code after a _cargo update_.

## Reproduce
With `docker run --rm -d -p 5672:5672 rabbitmq` running.

```rust
// [dependencies]
// lapin = "=1.6.9"
// tokio = { version = "1", features = ["full"] }

use lapin::{options::ExchangeDeclareOptions, Connection};

#[tokio::main]
async fn main() {
    let connection = Connection::connect("amqp://127.0.0.1:5672/%2f", <_>::default())
        .await
        .unwrap();

    let channel = connection.create_channel().await.expect("channel1");

    // expect this to fail as no such exchange exists
    // we also expect the channel to be dead now
    channel
        .exchange_declare(
            "foo-exchange",
            <_>::default(),
            ExchangeDeclareOptions {
                passive: true,
                ..<_>::default()
            },
            <_>::default(),
        )
        .await
        .unwrap_err();

    let _new_channel = connection.create_channel().await.expect("channel2");

    // non-passively declare exchange etc
}
```

### lapin = "=1.6.9"
```
thread 'main' panicked at 'channel2: InvalidChannelState(Error)', src/main.rs:31:58
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

### lapin = "=1.6.8"
No error.

**Comment by Keruspe** [comment]
Thanks a lot, will give it a look and add some regression test around this.

(I didn't know of <_>::default(), always used Default::default() :D)

**Comment by Keruspe** [comment]
Caused by 043704fb724d24152cb675808e87f129e2f1d51e

Didn't dig further yet but I probably know why (there's now a place where we set as error instead of just closing)

**Comment by Keruspe** [comment]
Fixed in 1.7.0

**Comment by alekspickle** [comment]
I'm also getting this `InvalidChannelState(Error)` error, but can't quite pinpoint it, any ideas?
